tfjs-backend-webgpu/src/kernels/Draw.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use backend file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {KernelConfig, KernelFunc, TensorInfo} from '@tensorflow/tfjs-core';
+import {Draw, DrawAttrs, DrawInputs,} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {ToPixelsProgram} from '../to_pixels_webgpu';
+
+import {DrawTextureGraphicsProgram} from '../webgpu_graphics_program';
+
+export function draw(
+    args: {inputs: DrawInputs, backend: WebGPUBackend, attrs: DrawAttrs}):
+    TensorInfo {
+  const {inputs, backend, attrs} = args;
+  const {image} = inputs;
+  const {canvas, options} = attrs;
+  const [height, width] = image.shape.slice(0, 2);
+  const {imageOptions} = options || {};
+  const alpha = imageOptions ?.alpha || 1;
+
+  const format = 'rgba8unorm';
+  const outShape = [height, width];
+  const useAlpha = alpha !== 1;
+  const program = new ToPixelsProgram(outShape, image.dtype, format, useAlpha);
+  canvas.width = width;
+  canvas.height = height;
+  const gpuContext = canvas.getContext('webgpu');
+  if (!gpuContext) {
+    throw new Error(
```

Comment:
If `webgpu` context is null, please check whether we can get a `2d` context. We can create an intermediate webgpu canvas and render to it. Then use `2d` canvas's drawImage to draw webgpu canvas to 2d canvas.